tests/unit/test_avro_compatibility.py

```diff
@@ -442,20 +454,22 @@ def test_union_to_simple_comparison(field):
     )
 )
 EMPTY_UNION_SCHEMA = UnionSchema([])
-NULL_UNION_SCHEMA = UnionSchema([NULL_SCHEMA])
-INT_UNION_SCHEMA = UnionSchema([INT_SCHEMA])
-LONG_UNION_SCHEMA = UnionSchema([LONG_SCHEMA])
-FLOAT_UNION_SCHEMA = UnionSchema([FLOAT_SCHEMA])
-DOUBLE_UNION_SCHEMA = UnionSchema([DOUBLE_SCHEMA])
-STRING_UNION_SCHEMA = UnionSchema([STRING_SCHEMA])
-BYTES_UNION_SCHEMA = UnionSchema([BYTES_SCHEMA])
-INT_STRING_UNION_SCHEMA = UnionSchema([INT_SCHEMA, STRING_SCHEMA])
-STRING_INT_UNION_SCHEMA = UnionSchema([STRING_SCHEMA, INT_SCHEMA])
-INT_FLOAT_UNION_SCHEMA = UnionSchema([INT_SCHEMA, FLOAT_SCHEMA])
-INT_LONG_UNION_SCHEMA = UnionSchema([INT_SCHEMA, LONG_SCHEMA])
-INT_LONG_FLOAT_DOUBLE_UNION_SCHEMA = UnionSchema([INT_SCHEMA, LONG_SCHEMA, FLOAT_SCHEMA, DOUBLE_SCHEMA])
-NULL_INT_ARRAY_UNION_SCHEMA = UnionSchema([NULL_SCHEMA, INT_ARRAY_SCHEMA])
-NULL_INT_MAP_UNION_SCHEMA = UnionSchema([NULL_SCHEMA, INT_MAP_SCHEMA])
+NULL_UNION_SCHEMA = UnionSchema([NULL_SCHEMA.to_json()], Names())
```

Comment:
Question: why is the `to_json` needed here?

Reply:
For example the UnionSchema, the Python 3 compatibility in Avro 1.10.x has this constructor: https://github.com/apache/avro/blob/release-1.10.2/lang/py3/avro/schema.py#L820
And on Python 2.x it uses this: https://github.com/apache/avro/blob/release-1.10.2/lang/py/avro/schema.py#L769

On the 1.11.0 the Python 3 layer is removed and code follows the previous 2.x logic.

Reply:
ugh, `to_json` doesn't return json, it returns the corresponding python object.